[DOCS][NFC] Update v6.22 release notes (#5545)
root-project/root: README/ReleaseNotes/v622/index.md

- `ROOT::GetImplicitMTPoolSize` has been deprecated in favor of the newly added `ROOT::GetThreadPoolSize` and
  will be removed in v6.24.
- Manually setting `TThreadedObject::fgMaxSlots` is deprecated: TThreadedObject now increases the number of slots
  on-demand rather than running out and throwing an exception

## Core Libraries

- The `ACLiC` can be configured to pass options to the `rootcling` invocation by enabling in the `.rootrc` the `ACLiC.ExtraRootclingFlags [-opts]` line.
- A call to `ROOT::EnableThreadSafety` is not required before using `TThreadExecutor` or `TTreeProcessorMT` anymore
- `TTreeProcessorMT` does not silently activate implicit multi-threading features anymore. An explicit call to
  `ROOT::EnableImplicitMT` is required instead

> - `TTreeProcessorMT` now accepts the number of threads its thread-pool should contain as constructor argument

Reviewer: Maybe
```suggestion
- `TTreeProcessorMT` now has a constructor argument to set the number of threads for its thread-pool
```

  and skipping the file, it now throws an exception if any of the input files is unreadable (this could also happen in
  the middle of an event loop). See [ROOT-10549](https://sft.its.cern.ch/jira/browse/ROOT-10549) for more details.
- New analysis examples based on the recent ATLAS Open Data release ([`Higgs to two photons`](https://root.cern/doc/master/df104__HiggsToTwoPhotons_8py.html), [`W boson analysis`](https://root.cern/doc/master/df105__WBosonAnalysis_8py.html), [`Higgs to four leptons`](https://root.cern/doc/master/df106__HiggsToFourLeptons_8py.html))
- An exception is now thrown in case the size of ROOT's thread-pool changes between RDataFrame construction time and the time the event loop begins.

> - Just-in-time compilation of large portions of the computation graph has been optimized, and it is now much faster. Please report any regressions you might encounter.

Reviewer: Maybe add on ROOT's jira / website / github?

Author: what do you mean?

Reviewer: I thought one could include the preferred place to report.

Author: ah yes of course!

Author: done

> - `MakeRootDataFrame` is now a safe way to construct RDFs. It used to return RDFs with more limited functionality.

Reviewer: ```suggestion
- `MakeRootDataFrame` now constructs more reliable RDataFrames. It used to return RDFs with limited functionality.
```

Author: I like my stronger wording better sorry :smile: If people were using `MakeRootDataFrame` in their code, they should be scared.
